web-client/public/js/update-app.js

```diff
@@ -393,7 +393,12 @@ const loadExpressionDatabase = function (isTopDataset) {
         queryExpressionDatabase({
             type:"ExpressionData",
             dataset,
-            genes : grnState.workbook.genes.map(gene => {return gene.name;}).join(","),
+            genes : grnState.workbook.genes.map(gene => {
+                if (grnState.workbook.meta.data.workbookType === NETWORK_PPI_MODE) {
+                    return gene.name.endsWith("p") ? gene.name.slice(0, -1) : gene.name;
+                }
+                return gene.name;
```

Comment:
In the same spirit as what we’re doing in _graph.js_, we can make this a 100% conditional expression as well:
```suggestion
                return (
                  grnState.workbook.meta.data.workbookType === NETWORK_PPI_MODE &&
                  gene.name.endsWith("p")
                ) ? gene.name.slice(0, -1)
                  : gene.name;
```

Reply:
Note also that upon unifying the coding style of the calculation in both _graph.js_ and _update-app.js_, we now see that the code is similar enough to extract as its own function. I won’t go into how that function exactly will be here, but it will probably involve some parameterization; however, doing that will still unify some of the logic

So, next step—recommendation is to do the first wave of changes, then test + commit; then when this approach looks good, make the next change of extracting a function from it

Reply:
Fixed